showkase-processor/src/main/java/com/airbnb/android/showkase/processor/writer/ShowkaseBrowserPropertyWriter.kt

```diff
@@ -0,0 +1,214 @@
+            showkaseMetadataCodeblock(showkaseMetadata)
+            addLineBreak()
+            add(")")
+        }
+            .build()
+    )
+        .addOriginatingElement(showkaseMetadata.element)
+        .build()
+
+    private fun getPropertyForComponentWithoutParameter(
+        propertyName: String,
+        showkaseMetadata: ShowkaseMetadata.Component
+    ): PropertySpec {
+        val property = PropertySpec.builder(
+            propertyName,
+            ShowkaseBrowserWriter.SHOWKASE_BROWSER_COMPONENT_CLASS_NAME
+        )
+            .initializer(
+                CodeBlock.Builder().apply {
+                    addShowkaseBrowserComponent(showkaseMetadata)
+                    addLineBreak()
+                    add(")")
+                }
+                    .build()
+            )
+            .addOriginatingElement(showkaseMetadata.element)
+            .build()
+        return property
+    }
+
+    private fun getPropertyForComponentWithParameter(
+        propertyName: String,
+        showkaseMetadata: ShowkaseMetadata.Component
+    ): PropertySpec {
+        return PropertySpec.builder(
+            propertyName,
+            List::class
+                .asTypeName()
+                .parameterizedBy(ShowkaseBrowserWriter.SHOWKASE_BROWSER_COMPONENT_CLASS_NAME)
+        ).apply {
+            initializer(
+                CodeBlock.Builder().apply {
+                    addPreviewProviderComponent(showkaseMetadata)
+                }
+                    .build()
+            )
+        }
+            .addOriginatingElement(showkaseMetadata.element)
+            .build()
+    }
+
+    private fun generatePropertyNameFromComponentMetadata(
+        showkaseMetadata: ShowkaseMetadata.Component,
+    ): String {
+        val name =
+            if (showkaseMetadata.componentIndex != null && showkaseMetadata.componentIndex > 0
+            ) {
+                "${showkaseMetadata.packageName}_${showkaseMetadata.showkaseGroup}_" +
+                        "${showkaseMetadata.showkaseName}_${showkaseMetadata.componentIndex}"
+            } else {
+                "${showkaseMetadata.packageName}_${showkaseMetadata.showkaseGroup}_${showkaseMetadata.showkaseName}"
+            }
+        val propertyName = if (showkaseMetadata.showkaseStyleName != null) {
+            "${name}_${showkaseMetadata.showkaseStyleName}"
+        } else {
+            name
+        }.filter { it.isLetterOrDigit() }
+        return propertyName
+    }
+
+    private fun generatePropertyNameFromMetadata(colorMetadata: ShowkaseMetadata) =
+        "${colorMetadata.packageName}_${colorMetadata.showkaseGroup}_${colorMetadata.showkaseName}"
+            .filter { it.isLetterOrDigit() }
+
+    private fun FileSpec.Builder.addPropertyAndGenerateFile(
+        propertySpec: PropertySpec,
+    ) {
+        addProperty(propertySpec)
+        build()
+            .writeTo(environment.filer, mode = XFiler.Mode.Aggregating)
```

Comment:
@vinaygaba I think you want to use isolating mode to take advantage of splitting the properties into different files and better incremental compilation (which was one of the goals of this change)

Reply:
https://kotlinlang.org/docs/ksp-incremental.html#aggregating-vs-isolating